template/src/ontology/Makefile.jinja2

```diff
@@ -141,6 +141,47 @@ config_check:
 $(TMPDIR) $(REPORTDIR) $(MIRRORDIR) $(IMPORTDIR) $(COMPONENTSDIR) $(SUBSETDIR):
 	mkdir -p $@
 
+# ----------------------------------------
```

Comment:
Why did you choose to not have `{% if project.robot_plugins is defined %}` wrapping the entire section? Few people will need robot custom plugins, so there is an argument here to be made about makefile hygiene. (a weak argument).

Reply:
Because we still need most of this section even if no custom plugins are defined in the spec, to install (1) the plugins that are provided by the ODK and (2) any plugin found in the top-level `plugins` directory.

Reply:
I am thinking about the case where the user does not want any of this (custom plugins to be installed from any source). My assumption is that this is 80% of the  ODK setups. In this case, there is really no reason for any extra line of code in the Makefile, no?

Your argument seems to be that "independent of what the user configures, if they put plugins in their user dir for plugins, they should be able to use the ODK makefile to pull them into the project". I think that any plugin the user intends to use should be mentioned in the config file, which means that unless `{% if project.robot_plugins is defined %}` there is no need to provide any code that allows handling plugins. But maybe I am not seeing something here!

Reply:
> I am thinking about the case where the user does not want any of this (custom plugins to be installed from any source). My assumption is that this is 80% of the ODK setups. In this case, there is really no reason for any extra line of code in the Makefile, no?

What about the plugins provided by the ODK (such as, for now, the SSSOM plugin)?

We want those plugins to be readily usable without users having to do anything special – that’s kind of the entire point of bundling them in the ODK, otherwise we would just ask people who want to use those plugins to configure them in the `plugins` section.

As for the top-level `plugins` directory, I believe it provides useful flexibility that would be more cumbersome to achieve if the only way to use custom plugins was through the config file. For example:

1) What if I want to _temporarily_ override a ODK-provided plugin (e.g., for testing or debugging purposes, I want to use a specific version of `sssom.jar` that is different from the one provided in the ODK)? All I have to do is to put the version I want in the top-level `plugins` directory (and remove it when I no longer need it). If the only way to install extra plugins was through the ODK config file, I would need to either (a) change the configuration and re-generate the Makefile, or (b) override the generated rule to install the plugin in the custom Makefile. Doable, sure, but more cumbersome.

2) What if I wrote a very specialised plugin that only makes sense for the particular ontology I am working on? I can simply put it into the top-level `plugins` folder (where I can optionally commit it to the repository). If the only way to install an extra plugin was through the ODK configuration file, I would need to either make sure the plugin is downloadable from somewhere, or to have a custom rule that copies it from somewhere on the local disk to the runtime plugins directory. Doable again, but again more cumbersome.

Reply:
As for the “extra lines of code in the Makefile”: those lines would only ever get executed _if a rule declares itself as being dependent on the `all_robot_plugins` target_ (be it a standard rule from the generated Makefile or a custom rule). If your workflow does not include any such rule, those extra lines of code will never run.

Reply:
If you want _less_ flexibility, I’d rather remove the possibility of declaring extra plugins through the config file and only leave the possibility of having custom installation rules. The config file is the least flexible option of all, so better for it to go instead of making it the only option.

Reply:
Nope, I am fine with the argumentation. Lets say we want to deliver some standard functionality through a ROBOT plugin, there should be a way to deliver it regardless of wether the option is defined or not. Thanks for your explanations!